pyuavcan/transport/can/media/pythoncan/_pythoncan.py

```diff
@@ -16,42 +24,174 @@
+    def __init__(self, iface_name: str, channel_name: str, bitrate: int, mtu: int) -> None:
+        self._iface_name = str(iface_name)
+        self._channel_name = str(channel_name)
+        self._mtu = int(mtu)
+        self._is_fd = self._mtu > _NativeFrameDataCapacity.CAN_CLASSIC
+        self._loop = asyncio.get_event_loop()
+        self._closed = False
+        self._maybe_thread: typing.Optional[threading.Thread] = None
+        self._background_executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+        self._loopback_enabled = False
+        self._bus = can.ThreadSafeBus(interface=iface_name, channel=channel_name, bitrate=bitrate)
+        self._loopback_lock = threading.RLock()
+        self._loop_frames = []
+        super(PythonCANMedia, self).__init__()
 
     @property
     def loop(self) -> asyncio.AbstractEventLoop:
-        raise NotImplementedError
+        return self._loop
 
     @property
     def interface_name(self) -> str:
-        raise NotImplementedError
+        return self._iface_name
 
     @property
     def mtu(self) -> int:
-        raise NotImplementedError
+        return self._mtu
 
     @property
     def number_of_acceptance_filters(self) -> int:
-        raise NotImplementedError
+        #just a placeholder to avoid error
+        return 16
 
     def start(self, handler: _media.Media.ReceivedFramesHandler, no_automatic_retransmission: bool) -> None:
-        raise NotImplementedError
+        if self._maybe_thread is None:
+            self._maybe_thread = threading.Thread(target=self._thread_function,
+                                                  name=str(self),
+                                                  args=(handler,),
+                                                  daemon=True)
+            self._maybe_thread.start()
+            if no_automatic_retransmission:
+                _logger.info('%s non-automatic retransmission is not supported', self)
+        else:
+            raise RuntimeError('The RX frame handler is already set up')
 
     def configure_acceptance_filters(self, configuration: typing.Sequence[_media.FilterConfiguration]) -> None:
-        _logger.warning('%s FIXME: acceptance filter configuration is not yet implemented; please submit patches! '
-                        'Requested configuration: %s',
-                        self, ', '.join(map(str, configuration)))
+        if self._closed:
+            raise pyuavcan.transport.ResourceClosedError(repr(self))
+        filters = []
+        for f in configuration:
+            f_dict = {}
+            f_dict['can_id'] = f.identifier
+            f_dict['can_mask'] = f.mask
+            f_dict['extended'] = f.format == _media.FrameFormat.EXTENDED
+            filters.append(f_dict)
+        _logger.info('Acceptance filters activated: %s', ', '.join(map(str, configuration)))
+        self._bus.set_filters(filters)
+                        
+    def send_msg(self, msg: can.Message) -> None:
```

Comment:
Is a private method defined with double underscore? I see that private methods of _socketcan.py are defined with single underscore (_parse_native_frame for example). Is it a correct way here?

Reply:
It takes just one underscore: `_send_msg`, although in this specific case, if you also consider my comment about timeout, it's better to make a closure or just use a lambda, like, instead of:

```
await self._loop.run_in_executor(self._background_executor, self.send_msg, message)
```

You say something along the lines of:

```
await self._loop.run_in_executor(self._background_executor, lambda: self._bus.send(message, timeout=monotonic_deadline - self._loop.time()))
```

Reply:
Ok, I see, seems to be a better solution here